cloud_database.go

```diff
@@ -0,0 +1,1282 @@
+}
+
+// Get a instances.
+func (ins *cloudDatabaseInstances) Get(ctx context.Context, instanceID string) (*CloudDatabaseInstance, error) {
+	req, err := ins.client.NewRequest(ctx, http.MethodGet, databaseServiceName, ins.resourcePath(instanceID), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := ins.client.Do(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	var instances *CloudDatabaseInstance
+
+	if err := json.NewDecoder(resp.Body).Decode(&instances); err != nil {
+		return nil, err
+	}
+
+	return instances, nil
+}
+
+// Action with a instances.
+func (ins *cloudDatabaseInstances) Action(ctx context.Context, instanceID string, iar *CloudDatabaseAction) (*CloudDatabaseMessageResponse, error) {
+	req, err := ins.client.NewRequest(ctx, http.MethodPost, databaseServiceName, ins.resourceActionPath(instanceID), &iar)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := ins.client.Do(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	var dmr *CloudDatabaseMessageResponse
+
+	if err := json.NewDecoder(resp.Body).Decode(&dmr); err != nil {
+		return nil, err
+	}
+
+	return dmr, nil
+}
+
+// Delete  a instances.
+func (ins *cloudDatabaseInstances) Delete(ctx context.Context, instanceID string, idr *CloudDatabaseDelete) (*CloudDatabaseMessageResponse, error) {
+	req, err := ins.client.NewRequest(ctx, http.MethodDelete, databaseServiceName, ins.resourcePath(instanceID), &idr)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := ins.client.Do(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	var dmr *CloudDatabaseMessageResponse
+
+	if err := json.NewDecoder(resp.Body).Decode(&dmr); err != nil {
+		return nil, err
+	}
+
+	return dmr, nil
+}
+
+// CloudDatabase Node
+
+// List all Node.
+func (no *cloudDatabaseNodes) List(ctx context.Context, opts *CloudDatabaseListOption) ([]*CloudDatabaseNode, error) {
+	req, err := no.client.NewRequest(ctx, http.MethodGet, databaseServiceName, cloudDatabaseNodesResourcePath, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	AddParamsListOption(req, opts)
+
+	resp, err := no.client.Do(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	var data struct {
+		Nodes []*CloudDatabaseNode `json:"nodes"`
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
+		return nil, err
+	}
+
+	return data.Nodes, nil
+}
+
+// Create a new replica or secondary nodes
+func (no *cloudDatabaseNodes) Create(ctx context.Context, icr *CloudDatabaseNodeCreate) (*CloudDatabaseNode, error) {
+	req, err := no.client.NewRequest(ctx, http.MethodPost, databaseServiceName, cloudDatabaseNodesResourcePath, &icr)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := no.client.Do(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	var nodes *CloudDatabaseNode
+	if err := json.NewDecoder(resp.Body).Decode(&nodes); err != nil {
+		return nil, err
+	}
+	return nodes, nil
+}
```

Comment:
Need validate payload before sent it to API server

Reply:
I don't understand this. is all Payload already validated with struct when user use it?